allennlp/modules/seq2seq_decoders/decoder_net.py

```diff
@@ -0,0 +1,88 @@
+from typing import Tuple, Dict, Optional
+import torch
+from allennlp.common import Registrable
+
+class DecoderNet(torch.nn.Module, Registrable):
+    # pylint: disable=abstract-method
+    """
+    A ``DecoderNet`` is a ``Module`` that takes as input a result of ``Seq2SeqEncoder`` and returns a
+    new sequence of vectors.
+    Decoder should implement function to perform a decoding step.
```

Comment:
Please say which functions explicitly or just leave this line off. (A bit confusing as-is.)